### ISSUE TITLE
Record is not update after the rollBack

### ISSUE THREAD

**Issue by hirenbhut93** [yiisoft/yii2#13380]
### What steps will reproduce the problem?
$model = SomeModel::findOne($id);

```php
$transaction = Yii::$app->db->beginTransaction();
try {
    // some code writting .............
    $transaction->commit();
}
catch(\Exception $e) {
    $transaction->rollBack();

    //this is not update why?
    $model->status = 2;
    $model->update(false);
}
```

### What is the expected result?
Update the record after roll back 
### What do you get instead?
No record update after rollback. 
### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.9
| PHP version      | 5.6
| Operating system | Ubuntu

**Comment by samdark** [comment]
What's the database used?

**Comment by yapi68** [comment]
What about `$model->updateAttributes()` instead `$model->update()` ???

**Comment by hirenbhut93** [comment]
@samdark mysql 5.5

**Comment by cebe** [comment]
> What about `$model->updateAttributes()` instead `$model->update()` ???

that should not make a difference.

**Comment by hirenbhut93** [comment]
@cebe this issues raised when I used `$model->delete()` in try block.

**Comment by cebe** [comment]
@hirenbhut93 then it depends on where the actual exception is being thrown. If the delete query is successful and you throw an exception in afterDelete() the model is already not connected with the database row anymore, so the update is not possible.

**Comment by hirenbhut93** [comment]
@cebe I throw this exception when query give integrity error.

Any ideas for how to resolve this situation or how to do any alternative way.

**Comment by Alex-Code** [comment]
Does this work?
```php
\Yii::$app->db->createCommand()->update(SomeModel::tableName(), ['status' => 2], ['id' => $id])->execute();
```

**Comment by hirenbhut93** [comment]
@Alex-Code Yes its is work i knew, and also work when apply re initialize this model like...

```php
catch(\Exception $e) {
    $transaction->rollBack();

    // If apply re-initialize at time working....
    $model = SomeModel::findOne($id);
    $model->status = 2;
    $model->update(false);
}
```

**Comment by andrewnester** [comment]
@hirenbhut93 could please describe what exactly exception you are receiving when call to `$model->delete()`? Thanks!

**Comment by hirenbhut93** [comment]
@andrewnester `Exception (Integrity constraint violation) &#039;yii\db\IntegrityException&#039; with message &#039;SQLSTATE[23000]: Integrity constraint violation: 1451 Cannot delete or update a parent row: a foreign key constraint fails`

**Comment by andrewnester** [comment]
@hirenbhut93 are you sure there is no any logic in `afterDelete` method for you model? something like deleting related models or so?

**Comment by hirenbhut93** [comment]
@andrewnester Yes, I have used after delete logic, even though i have tested without after delete method, but my issue is not resolved.

**Comment by terabytesoftw** [comment]
Thanks for reporting this. I re-checked the behavior on current Yii2 master, (March 2026) with MySQL.

If `delete()` fails with an `IntegrityException`, rolling back and then calling `update(false)` on the same model instance works.

The “no update after rollback” case happens when `delete()` succeeded first and rollback occurs later. This is expected ActiveRecord behavior: after a successful `delete()`, the model is marked as new (old attributes are cleared), and transaction rollback does not restore in-memory AR object state.

Re-fetching the model (`findOne()`/`refresh()`) before updating works as expected.
